lib/engine/game/g_1837/game.rb

```diff
@@ -770,11 +770,16 @@ def token_graph_for_entity(_entity)
         end
 
         def upgrades_to?(from, to, special = false, selected_company: nil)
-          return yellow_town_tile_upgrades_to?(from, to) if from.color == :yellow && !from.towns.empty?
+          return yellow_town_tile_upgrades_to?(from, to) if use_yellow_town_tile_upgrades_to?(from)
 
           super
         end
 
+        # Need this separate method due to 1824
+        def use_yellow_town_tile_upgrades_to?(from)
```

Comment:
Why did yo need this method instead of overwriting `yellow_town_tile_upgrades_to?` to the proper implementation for 24?

Reply:
But you are right. Implementing yellow_town_tile_upgrades_to? was simple. Just like 1837 that method do not need to check for color progression, but just that existing paths are respected.
So I have changed this to remove the change in 1837, and do it in 1824 only.